addons/binding/org.openhab.binding.wifiled/src/main/java/org/openhab/binding/wifiled/handler/ClassicWiFiLEDDriver.java

```diff
@@ -149,16 +170,28 @@ private void sendLEDData(LEDStateDTO ledState) throws IOException {
             byte w = (byte) (((int) (ledState.getWhite().doubleValue() * 255 / 100)) & 0xFF);
             byte w2 = (byte) (((int) (ledState.getWhite2().doubleValue() * 255 / 100)) & 0xFF);
 
-            byte[] bytes = getBytesForColor(r, g, b, w, w2);
+            bytes = getBytesForColor(r, g, b, w, w2);
 
-            sendRaw(bytes);
         } else {
             // program selected
             byte p = (byte) (program & 0xFF);
             byte s = (byte) (((100 - ledState.getProgramSpeed().intValue()) * 0x1F / 100) & 0xFF);
-            byte[] data = { 0x61, p, s };
-            sendRaw(data);
+            bytes = new byte[]{ 0x61, p, s };
         }
+
+        ledUpdateFuture = updateScheduler.submit(() -> {
+            try {
+                Thread.sleep(WAIT_UPDATE_LED_FOR_MS);
+                logger.debug("Setting LED State to {}", bytesToHex(bytes));
+                sendRaw(bytes);
+                return true;
+            } catch (IOException e) {
+                logger.warn("IOException", e);
```

Comment:
Martin,

this item I was meaning to ask what the best way to handle this:

1) Since this run's in a separate thread I cannot send a response back to the handler. How is this normally handled within openhab?

2) I am not sure if it's the right way to directly update the item state, I assume you mean here to put it offline? What I see here locally that I often get  failed transmissions, This can be due to wifi or that they are just crappy devices... Is it ok to set a item offline and back online quickly after and how should retry be handled?